### ISSUE TITLE
DATAMATRIX corruption on certain strings

### ISSUE THREAD

**Issue by andybingham** [tecnickcom/tc-lib-barcode#48]
We generate DATAMATRIX codes strings, e.g.
(400)BS2WZ64PA(00)0
(400)BS2WZ64QA(00)0
The first scans as (400)BS2WZ64V#CB'A<\AW
the second scan fine.

Any ideas where the library is corrupting the string (and why?)
Using an online generator, both of these strings are fine, and the image for the first is different from the one generated by this library.

**Comment by andybingham** [comment]
![test](https://user-images.githubusercontent.com/17315894/67197169-1ddb6c80-f3f4-11e9-9352-ad303994af33.png)
This is the PNG generated by the first string

**Comment by andybingham** [comment]
I have found a work around, if  you force the string variable to UTF-16 it seems to encode the string OK.  I have not tested to see whether other string then corrupt.

**Comment by andybingham** [comment]
I have now finished testing with 1000 random generated string, using ASCII or UTF-8 encoding approx 1-2% fail to scan the correct string.  Using UTF-16, all work fine.
This does give us a work around for now (force the string to UTF-16, but there is certainly something wrong with the encoding process.

**Comment by gregglahti** [comment]
We're having the issue with this particular string:

```
LD2B 1 CLNGP
```

UTF-16 wasn't an option for us that worked.  Instead, what we've done to temporarily resolve this issue for us is to remove the the loook-ahead test to check for another encoding mode which then switches to that particular encoding mode.  That seemed to be the root issue for us when our string above got represented into the image when scanned as:

```
LD2B 1 CLNGPGQPd
```

Our change details:

In src/Type/Square/Datamatrix/Encode.php, line 70:

```
            if ($newenc != $enc) {
```

replaced with:

```
            if (FALSE) {
```

I'm a bit confused on why the check to switch the stream encoding in the middle of the translation is present, specifically since the Datamatrix is pretty much hard-fixed to ENC_ASCII (Datamatrix.php, line 188).  In our case, pretty much everything going in is ASCII.  

Your mileage may vary.

**Comment by nicolaasuni** [comment]
Thank you for reporting these issues, I have fixed them in the latest version (1.15.17).
